src/main/java/org/apache/mesos/hdfs/config/SchedulerConf.java

```diff
@@ -1,10 +1,28 @@
 package org.apache.mesos.hdfs.config;
 
+import com.google.inject.Singleton;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
 
-public class SchedulerConf extends MainConf {
-  public SchedulerConf(Configuration conf, int configServerPort) {
-    super(conf, configServerPort);
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Properties;
+
+@Singleton
+public class SchedulerConf extends Configured {
+
+  public SchedulerConf(Configuration conf) {
+    setConf(conf);
+  }
+
+  public SchedulerConf() {
+    Properties props = System.getProperties();
```

Comment:
Instead of reading from another properties file (System.properties), let's make this even cleaner and simply use the path etc/hadoop/mesos-site.xml directly. It will always be on that path as part of the framework.